src/main/java/fr/ens/biologie/genomique/eoulsan/modules/diffana/DESeq2Checker.java

```diff
@@ -60,22 +89,77 @@ static boolean checkExperimentDesign(Experiment experiment, boolean throwsExcept
                 && !getAllSamplesMetadataKeys(experiment.getDesign()).contains(REP_TECH_GROUP_KEY)) {
             return error("RepTechGroup column missing for experiment: "
                     + experiment.getName(), throwsException);
         }
 
+        /* Check if there is no numeric character at the begin of a row in the column Condition
+        for a complex design model */
+        if (getExperimentSampleAllMetadataKeys(experiment).contains(CONDITION_KEY)) {
+            for (ExperimentSample es : experiment.getExperimentSamples()) {
+                String s = DesignUtils.getMetadata(es, CONDITION_KEY);
+                // Condition column contains an invalid numeric character as first character
+                if ( Character.isDigit(s.charAt(0)) && emd.getComparisons() != null){
+                    return error("One or more Condition rows start with a numeric character : "
+                            + experiment.getName(), throwsException);
+                }
+            }
+        }
+
+        // Check if there is no "-" in the column Condition when the contrast mode is activate
+        if (getExperimentSampleAllMetadataKeys(experiment).contains(CONDITION_KEY)) {
+            for (ExperimentSample es : experiment.getExperimentSamples()) {
+                String s = DesignUtils.getMetadata(es, CONDITION_KEY);
+                if (s.contains("-") && emd.isContrast()){
+                    return error("There is a - character in the column Condition : "
+                            + experiment.getName(), throwsException);
+                }
+            }
+        }
+
+        // Verify consistency between the values in the columns Reference and Condition for non complex mode
+        // If mode is not complex
+        if (!emd.containsComparisons()) {
+            // If Exp.exp1.Condition and Exp.exp1.Reference exist
+            if (getExperimentSampleAllMetadataKeys(experiment).contains(REFERENCE_KEY) && getExperimentSampleAllMetadataKeys(experiment).contains(CONDITION_KEY)){
+                Map<String, String> lhm = new LinkedHashMap<>();
+                // Get all values of the Condition and Reference columns
+                for (ExperimentSample es : experiment.getExperimentSamples()) {
+                    String conditions = DesignUtils.getMetadata(es, CONDITION_KEY);
+                    String references = DesignUtils.getMetadata(es, REFERENCE_KEY);
+                    // If one condition is associated with more than one reference, error
+                    ArrayList<String> possibleConditionsReferences = new ArrayList<String>();
+                    possibleConditionsReferences.add(conditions+references);
+                    // if condition already in dictionnary check if value is equals to reference and if not error
+                    for (String key : lhm.keySet()) {
+                        String value = lhm.get(key);
+                        if (key.equals(conditions) && !value.equals(references)){
```

Comment:
Use the following code instead to get the key and the value at the same time:
`for (Map.Entry<String, String> e : lhm.entrySet())`
And then, use e.getKey() and e.getValue() to get the key and value.